internal/pipe/webhook/webhook.go

```diff
@@ -0,0 +1,158 @@
+package webhook
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/caarlos0/env/v6"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+const (
+	defaultMessageTemplate = `{ "message": "{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .ReleaseURL }}"}`
+	ContentTypeHeaderKey   = "Content-Type"
+	UserAgentHeaderKey     = "User-Agent"
+	UserAgentHeaderValue   = "gorleaser"
+	AuthorizationHeaderKey = "Authorization"
+	DefaultContentType     = "application/json; charset=utf-8"
+
+	EndpointRegex = "(http|nats)(s?)://.*"
```

Comment:
why nats?

Reply:
Its a very good cloud native eventing. NATS trigger are very common and its not hurting to support it. In case people use NATS as backbone for their Serverless landscape. Many reasons.

Reply:
NATS trigger are dope? it does not hurt to let it in and I wanted to use for an article when this feature is out :)?

Reply:
I would say maybe its better to not even check the protocol via regex and let the user do whatever? 

if it doesn't work, it'll fail anyway... this way we don't need to at some point add other protocols 😁

Reply:
Sounds good! I removed it! :)